From 95b5a08e3a714e1c21a9612284460f7febe29876 Mon Sep 17 00:00:00 2001
From: Ram Kumar Dwivedi <quic_rdwivedi@quicinc.com>
Date: Thu, 21 Aug 2025 16:54:02 +0530
Subject: [PATCH] arm64: dts: qcom: sm8650: Enable MCQ support for UFS
 controller

Enable Multi-Circular Queue (MCQ) support for the UFS host controller
on the Qualcomm SM8650 platform by updating the device tree node. This
includes adding new register region for MCQ and specifying the MSI parent
required for MCQ operation.

Signed-off-by: Ram Kumar Dwivedi <quic_rdwivedi@quicinc.com>
Source: https://lore.kernel.org/all/20250821112403.12078-5-quic_rdwivedi@quicinc.com/
Local adaptation: rebased context around dma-coherent in Linux 7.3-rc5;
register ranges, names and MSI parent are unchanged.

---
 arch/arm64/boot/dts/qcom/sm8650.dtsi | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/sm8650.dtsi b/arch/arm64/boot/dts/qcom/sm8650.dtsi
index b679fd8af..749211c4d 100644
--- a/arch/arm64/boot/dts/qcom/sm8650.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8650.dtsi
@@ -4041,7 +4041,10 @@ ufs_mem_phy: phy@1d80000 {
 
 		ufs_mem_hc: ufshc@1d84000 {
 			compatible = "qcom,sm8650-ufshc", "qcom,ufshc", "jedec,ufs-2.0";
-			reg = <0 0x01d84000 0 0x3000>;
+			reg = <0 0x01d84000 0 0x3000>,
+			      <0 0x1da0000 0 0x15000>;
+			reg-names = "std",
+				    "mcq";
 
 			interrupts = <GIC_SPI 265 IRQ_TYPE_LEVEL_HIGH 0>;
 
@@ -4079,6 +4082,8 @@ &mc_virt SLAVE_EBI1 QCOM_ICC_TAG_ALWAYS>,
 
 			iommus = <&apps_smmu 0x60 0>;
 
+			msi-parent = <&gic_its 0x60>;
+
 			dma-coherent;
 
 			lanes-per-direction = <2>;
-- 
2.55.0

